cjdns-admin/src/func_ret.rs

```diff
@@ -0,0 +1,83 @@
+//! Remote function return values.
+
+use std::collections::BTreeMap;
+use std::fmt;
+
+use serde::{Deserialize, Deserializer};
+use serde::de::{Error, MapAccess, SeqAccess, Unexpected, Visitor};
+
+/// Remote function return value. Supports json-like data types.
+#[derive(Clone, PartialEq, Eq)]
+pub enum ReturnValue {
+    /// Integer return value.
+    Int(i64),
+    /// String return value.
+    String(String),
+    /// List return value.
+    List(Vec<ReturnValue>),
+    /// Map return value.
+    Map(BTreeMap<String, ReturnValue>),
+}
+
+impl<'de> Deserialize<'de> for ReturnValue {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        deserializer.deserialize_any(ReturnValueVisitor)
+    }
+}
+
+struct ReturnValueVisitor;
+
+impl<'de> Visitor<'de> for ReturnValueVisitor {
```

Comment:
What is it for? Small comment would be nice.

Reply:
Just a standard serde stuff (manual `Deserialize` trait implementation), pretty obvious to be commented imho.